src/ansys/dpf/core/data_sources.py

```diff
@@ -183,9 +260,9 @@ def guess_second_key(filepath: str) -> str:
         return new_key
 
     def set_domain_result_file_path(
-        self, path: Union[str, os.PathLike], domain_id: int, key: Union[str, None] = None
-    ):
-        """Add a result file path by domain.
+        self, path: Union[str, os.PathLike], domain_id: int, key: Optional[str] = None
```

Comment:
```suggestion
        self, path: Union[str, os.PathLike], domain_id: int, key: Optional[str] = None
```
I think here `None` has to be listed as a valid type for the key argument.

Reply:
I think when we use optional is exactly to omit the "None" in a union